common/recordCreate.js

```diff
@@ -0,0 +1,308 @@
+         */
+        function checkUpdate(submissionRowsCopy, rootScope){
+            /**
+             * After uploading files, the returned submissionRowsCopy contains
+             * new file data. This includes filename, filebyte, and md5.
+             * The following makes sure that all the data are updated.
+             * That's why this for loop must be after uploading files and not before.
+             * And we cannot just pass submissionRowsCopy to update function, because
+             * update function only accepts array of tuples (and not just key-value pair).
+             */
+            for (var i = 0; i < submissionRowsCopy.length; i++) {
+                var row = submissionRowsCopy[i];
+                var data = rootScope.tuples[i].data;
+                // assign each value from the form to the data object on tuple
+                for (var key in row) {
+                    data[key] = (row[key] === '' ? null : row[key]);
+                }
+            }
+            return data;
+        }
+
+        /**
+         * uploadFiles - uploading files
+         *
+         * @param  {array} submissionRowsCopy   data that is going to be uploaded
+         * @param  {object} rootScope           object contains global values from calling function
+         * @param  {object} onSuccess           callback
+         */
+        function uploadFiles(submissionRowsCopy, rootScope, onSuccess) {
+
+                $uibModal.open({
+                    templateUrl: "../common/templates/uploadProgress.modal.html",
+                    controller: "UploadModalDialogController",
+                    controllerAs: "ctrl",
+                    size: "md",
+                    backdrop: 'static',
+                    keyboard: false,
+                    resolve: {
+                        params: {
+                            reference: rootScope.reference,
+                            rows: submissionRowsCopy
+                        }
+                    }
+                }).result.then(onSuccess, function(exception) {
+                    viewModel.readyToSubmit = false;
+                    viewModel.submissionButtonDisabled = false;
+
+                    if (exception) AlertsService.addAlert(exception.message, 'error');
+                });
+        }
+
+        /**
+         * addRecords - Function that calls ermrestjs method to add data
+         *
+         * @param  {bool} isUpdate              flag
+         * @param  {object} derivedref          derived referene of the column that is being added
+         * @param  {array} recordEditModel      data array
+         * @param  {bool} isModalUpdate         if updating through record app
+         * @param  {object} rootScope           object contains global values from calling function
+         * @param  {object} onSuccessFunction   callback
+         */
+        function addRecords(isUpdate, derivedref, recordEditModel, isModalUpdate, rootScope, onSuccessFunction) {
```

Comment:
Same comment here as above. The whole rootscope is not needed, only the `tuples` and `reference`. Only pass those into this function as separate parameters.

Make sure these 2 values are not already present on `recordEditModel` though.

Reply:
I guess I didn't notice that the `queryParams` were being used as well (`rootscope.context.queryParams`). This should be another parameter to the `addRecords` function.